### ISSUE TITLE
"getStubWorkspace generated incompatible stub workspace"

### ISSUE THREAD

**Issue by NicholasFields** [nikitabobko/AeroSpace#184]
Context:

1. Plugged monitor directly into M3 MacBook Pro HDMI port
2. Noticed aerospace wasn't running (as my chrome window was still seemingly sized for the built-in screen)
3. Restarted aerospace and found the terminal with the error (below) on workspace 1

------

```
###############################
### AEROSPACE RUNTIME ERROR ###
###############################

Please report to:
    https://github.com/nikitabobko/AeroSpace/issues/new

Message: getStubWorkspace generated incompatible stub workspace (Workspace(name: '1', isVisible: 'false', isEffectivelyEmpty: 'true')) for the monitor ((1920.0, 50.0)
Version: 0.9.0-Beta
Git hash: f952a6a771b8b9dc2dd453e993a55f7f6d91b2ae
Coordinate: /Users/bobko/a/AeroSpace/src/tree/Workspace.swift:167:14 rearrangeWorkspacesOnMonitors()
recursionDetectorDuringFailure: false
cli: false

Stacktrace:
0   AeroSpace                           0x0000000104d98708 $s6Common6errorT_4file4line6column8functionxSS_SSS2iSStlFs5NeverO_Tg5 + 768
1   AeroSpace                           0x0000000104dad06c $s6Common5check__4file4line6column8functionySb_S2SS2iSStF + 56
2   AeroSpace                           0x0000000104cdae74 $s9AeroSpace29rearrangeWorkspacesOnMonitors33_6104020D6526819D1E9D3B2C30C9FD5BLLyyF + 1944
3   AeroSpace                           0x0000000104ce0f18 $s9AeroSpace7MonitorPAAE15activeWorkspaceAA0E0CvgTf4n_g + 168
4   AeroSpace                           0x0000000104cd5040 $s9AeroSpace14updateTrayTextyyFSSAA7Monitor_pXEfU1_ + 76
5   AeroSpace                           0x0000000104cd52f8 $sSlsE3mapySayqd__Gqd__7ElementQzKXEKlFSay9AeroSpace7Monitor_pG_SSTg503$s9c29Space14updateTrayTextyyFSSAA7E8_pXEfU1_AGTf1cn_nTf4ng_n + 112
6   AeroSpace                           0x0000000104cd4e3c $s9AeroSpace14updateTrayTextyyF + 200
7   AeroSpace                           0x0000000104d0d0e0 $s9AeroSpace14refreshSession7startup10forceFocus4bodyxSb_SbxyXEtlFyt_Tg503$s9A44Space16refreshAndLayout7startupySb_tFyyXEfU_Tf1nnnc_n + 1688
8   HIServices                          0x000000018d9d23f4 _XXMIGPostNotification + 888
9   HIServices                          0x000000018d9fd4a8 _XPostNotification + 360
10  HIServices                          0x000000018d9b43b8 mshMIGPerform + 204
11  CoreFoundation                      0x000000018765bfdc __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__ + 60
12  CoreFoundation                      0x000000018765befc __CFRunLoopDoSource1 + 520
13  CoreFoundation                      0x000000018765a8c8 __CFRunLoopRun + 2244
14  CoreFoundation                      0x00000001876599ac CFRunLoopRunSpecific + 608
15  HIToolbox                           0x0000000191c08448 RunCurrentEventLoopInMode + 292
16  HIToolbox                           0x0000000191c08284 ReceiveNextEventCommon + 648
17  HIToolbox                           0x0000000191c07fdc _BlockUntilNextEventMatchingListInModeWithFilter + 76
18  AppKit                              0x000000018ae368a4 _DPSNextEvent + 660
19  AppKit                              0x000000018b610980 -[NSApplication(NSEventRouting) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 716
20  AppKit                              0x000000018ae29d50 -[NSApplication run] + 476
21  AppKit                              0x000000018ae01014 NSApplicationMain + 880
22  SwiftUI                             0x00000001b22b4364 OUTLINED_FUNCTION_12 + 15824
23  SwiftUI                             0x00000001b2b4393c OUTLINED_FUNCTION_3 + 196
24  SwiftUI                             0x00000001b2f8ad60 OUTLINED_FUNCTION_1 + 152
25  AeroSpace                           0x0000000104ca8a4c main + 36
26  dyld                                0x00000001871fd0e0 start + 2360
```

----

Reporting as the error requests that I do so -- 
please LMK if you want more info but this was of no real consequence to me. Simply restarted and good to go.

Thanks for building Aerospace!

**Comment by NicholasFields** [comment]
Repro clarification:

1. Aerospace running on external monitor with MacBook lid closed
2. Open MacBook Lid (activates built-in screen)
3. Aerospace crashes and error message is offered up

---

Latest occurence's error message (haven't cheched thorougly but looks much the same):

```
###############################
### AEROSPACE RUNTIME ERROR ###
###############################

Please report to:
    https://github.com/nikitabobko/AeroSpace/issues/new

Message: getStubWorkspace generated incompatible stub workspace (Workspace(name: '2', isVisible: 'false', isEffectivelyEmpty: 'true')) for the monitor ((1920.0, 50.0)
Version: 0.9.0-Beta
Git hash: f952a6a771b8b9dc2dd453e993a55f7f6d91b2ae
Coordinate: /Users/bobko/a/AeroSpace/src/tree/Workspace.swift:167:14 rearrangeWorkspacesOnMonitors()
recursionDetectorDuringFailure: false
cli: false

Stacktrace:
0   AeroSpace                           0x0000000100c74708 $s6Common6errorT_4file4line6column8functionxSS_SSS2iSStlFs5NeverO_Tg5 + 768
1   AeroSpace                           0x0000000100c8906c $s6Common5check__4file4line6column8functionySb_S2SS2iSStF + 56
2   AeroSpace                           0x0000000100bb6e74 $s9AeroSpace29rearrangeWorkspacesOnMonitors33_6104020D6526819D1E9D3B2C30C9FD5BLLyyF + 1944
3   AeroSpace                           0x0000000100bbcf18 $s9AeroSpace7MonitorPAAE15activeWorkspaceAA0E0CvgTf4n_g + 168
4   AeroSpace                           0x0000000100bb1040 $s9AeroSpace14updateTrayTextyyFSSAA7Monitor_pXEfU1_ + 76
5   AeroSpace                           0x0000000100bb12f8 $sSlsE3mapySayqd__Gqd__7ElementQzKXEKlFSay9AeroSpace7Monitor_pG_SSTg503$s9c29Space14updateTrayTextyyFSSAA7E8_pXEfU1_AGTf1cn_nTf4ng_n + 112
6   AeroSpace                           0x0000000100bb0e3c $s9AeroSpace14updateTrayTextyyF + 200
7   AeroSpace                           0x0000000100be90e0 $s9AeroSpace14refreshSession7startup10forceFocus4bodyxSb_SbxyXEtlFyt_Tg503$s9A44Space16refreshAndLayout7startupySb_tFyyXEfU_Tf1nnnc_n + 1688
8   HIServices                          0x000000018d9d23f4 _XXMIGPostNotification + 888
9   HIServices                          0x000000018d9fd4a8 _XPostNotification + 360
10  HIServices                          0x000000018d9b43b8 mshMIGPerform + 204
11  CoreFoundation                      0x000000018765bfdc __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__ + 60
12  CoreFoundation                      0x000000018765befc __CFRunLoopDoSource1 + 520
13  CoreFoundation                      0x000000018765a8c8 __CFRunLoopRun + 2244
14  CoreFoundation                      0x00000001876599ac CFRunLoopRunSpecific + 608
15  HIToolbox                           0x0000000191c08448 RunCurrentEventLoopInMode + 292
16  HIToolbox                           0x0000000191c080d8 ReceiveNextEventCommon + 220
17  HIToolbox                           0x0000000191c07fdc _BlockUntilNextEventMatchingListInModeWithFilter + 76
18  AppKit                              0x000000018ae368a4 _DPSNextEvent + 660
19  AppKit                              0x000000018b610980 -[NSApplication(NSEventRouting) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 716
20  AppKit                              0x000000018ae29d50 -[NSApplication run] + 476
21  AppKit                              0x000000018ae01014 NSApplicationMain + 880
22  SwiftUI                             0x00000001b22b4364 OUTLINED_FUNCTION_12 + 15824
23  SwiftUI                             0x00000001b2b4393c OUTLINED_FUNCTION_3 + 196
24  SwiftUI                             0x00000001b2f8ad60 OUTLINED_FUNCTION_1 + 152
25  AeroSpace                           0x0000000100b84a4c main + 36
26  dyld                                0x00000001871fd0e0 start + 2360

```

**Comment by NicholasFields** [comment]
Ahh -- same thing when I attempt to run aerospace with both the external and built-in monitors active.
==> Immediate crash + the same error message.

Doesn't affect my use case (single montor-ing for now) but LMK if you need anything further.

**Comment by nikitabobko** [comment]
Thanks! Should be fixed in 0.9.1-Beta
